youtube.go

```diff
@@ -134,6 +142,39 @@ func (y *Youtube) StartDownloadFile() error {
 	return err
 }
 
+//  StartDownloadWithItag : Starting download video according to given itag value
+func (y *Youtube) StartDownloadWithItag(destFile string, itag int) error {
+	err := errors.New("Empty stream list")
+	itagFound := false
+	for _, stream := range y.StreamList {
+		if stream.Itag == itag {
+			itagFound = true
+			streamURL := stream.URL
+			y.log(fmt.Sprintln("Download url=", streamURL))
+
+			if destFile == "" {
+				// Find out what the file name should be.
+				fileName := SanitizeFilename(stream.Title)
+				fileName += pickIdealFileExtension(stream.Type)
+
+				usr, _ := user.Current()
+				destFile = filepath.Join(filepath.Join(usr.HomeDir, "Movies", "youtubedr"), fileName)
+			}
+
+			y.log(fmt.Sprintln("Download to file=", destFile))
+			err = y.videoDLWorker(destFile, streamURL)
+			if err == nil {
+				return nil
+			}
+		}
+	}
+
+	if !itagFound {
```

Comment:
one condition is needed :  len(y.StreamList)!=0 , otherwise we get itag invalid error by empty streamList

Reply:
I am not pretty sure if we need this check because we've verified that in the `DecodeURL` -> `parseVideoInfo` function. https://github.com/kkdai/youtube/blob/master/youtube.go#L271-L273 I was also wondering if we need to keep this error `err := errors.New("Empty stream list")`. maybe I misunderstand this part. any thought?

Reply:
No, you are right. You made good point,  I didnt' catch the error check in patseVideoInfo .  Using empty stream error could be redundant in all StartDownload methods, but not sure at this moment, because I think null check in the same method is also good.  Maybe I clean up the code later to get more information.

BTW, I have tested the code,  good job!